libpdbg/device.c

```diff
@@ -736,6 +736,56 @@ bool pdbg_targets_init(void *fdt)
 	return true;
 }
 
+/**
+ * @brief A local function to iterate through the target node for its children and delteing them
+ * from the list. Should be only called internally. Recursive in nature
+ * 
+ * @param target The target node for which the children will be reset
+ */
+
+static void pdbg_reset_children(struct pdbg_target *target)
+{
+	struct pdbg_target* childTraget = NULL;
+	enum pdbg_target_status status;
+
+	/* Does this target actually exist? */
+	if (target)
+	{
+    status = pdbg_target_status(target);
+		if (status != PDBG_TARGET_ENABLED)
+			return;
+	}
```

Comment:
General question: do we need to explicitly check if the target is a node?

Reply:
> General question: do we need to explicitly check if the target is a node?

Yes, because while probing the nodes after initiating it sets the nodes which are non existent and non functional as  PDBG_TARGET_DISABLED or PDBG_TARGET_RELEASED. Thus explicit checking is better to be on the safer side.